src/main/java/com/dev/cinema/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.dev.cinema.dao;
+
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.Movie;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```

Comment:
we should not use try with resources in create method coz if the transaction didn't happen, the session would close and we wouldn't go to `catch` and rollback wouldn't happen. So it's better to use try-catch-finally and close session in finally block